### PR TITLE
Fix editor's preview bug

### LINKED ISSUE #20701: Funniest bug ever
**Describe the bug**
Unable to publish

**To Reproduce**

Click on preview, and save disppear when hovering *trying* hover on it

**Expected behavior**

Ability to publish

**Screenshots**

https://github.com/forem/forem/assets/26553194/e90face3-4851-45ef-a584-1408d5ec7e10

**Desktop (please complete the following information):**

- OS, version: MacOS Sonoma 14.0 
- Browser, version: Thorium 116 and Safari 17

### PR DESCRIPTION
**Pull request by GwartneyDev**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Unable to publish article after going into preview mode.

When clicking on preview, and then going to save article the button disappears. 

As others have shown and stated in the issue reported when mousing over the bottom bar it triggers the switchHelpContext function, which in turn sets the state to the common Props which has the preview set to false.

As a suggested fix to this it was found that it either to pass in the current previewShowing state or prevent switchHelpContext from triggering if the preview is open. 

I have gone ahead and in a simple state check as shown here `if (!this.state.previewShowing)`. 

## Related Tickets & Documents

- Related Issue # https://github.com/forem/forem/issues/20701
- Closes #20701

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [ ] Semantic HTML implemented?
- [ ] Keyboard operability supported?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

### CODE REVIEW

**Comment by GwartneyDev** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by GwartneyDev** [comment]
recheck

File: app/javascript/article-form/articleForm.jsx [hunk 1]
```diff
@@ -409,14 +409,15 @@ export class ArticleForm extends Component {
...
   switchHelpContext = (event, override = null) => {
-    const id = override || event.target.id;
-    this.setState({
-      ...this.setCommonProps({
-        helpFor: id,
-        helpPosition: event.target.getBoundingClientRect().y,
-      }),
-    });
-  };
+    if (!this.state.previewShowing) {
```
Comment (eznix86) [inline app/javascript/article-form/articleForm.jsx:412]: I think this can be just an early return instead of wrapping it with an if.

Comment (GwartneyDev) [reply app/javascript/article-form/articleForm.jsx:412]: I did not think of that actually I think is a better way as well.

**Comment by GwartneyDev** [comment]
@lightalloy and @maestromac tagging for review follow up.

**Review by maestromac — APPROVED** [review]
Thanks for this! LGTM!

**Comment by GwartneyDev** [comment]
> Thanks for this! LGTM!

You are most welcome and thanks for the follow up.

**Comment by eznix86** [comment]
Yay! I can post now 😂

**Pull request merged**